modules/rostests/apitests/ntdll/DllLoadNotification.c

```diff
@@ -0,0 +1,225 @@
+    _In_opt_ PVOID Context,
+    _Out_ PVOID* Cookie);
+
+typedef
+NTSTATUS
+NTAPI
+FN_LdrUnregisterDllNotification(
+    _In_ PVOID Cookie);
+
+static BOOL ExtractResource(
+    _In_z_ PCWSTR SavePath,
+    _In_ PCWSTR ResourceType,
+    _In_ PCWSTR ResourceName)
+{
+    BOOL bSuccess;
+    DWORD dwWritten, dwSize;
+    HGLOBAL hGlobal;
+    LPVOID pData;
+    HANDLE hFile;
+    HRSRC hRsrc;
+
+    /* Load resource */
+    if ((hRsrc = FindResourceW(NULL, ResourceName, ResourceType)) == NULL ||
+        (dwSize = SizeofResource(NULL, hRsrc)) == 0 ||
+        (hGlobal = LoadResource(NULL, hRsrc)) == NULL ||
+        (pData = LockResource(hGlobal)) == NULL)
+    {
+        return FALSE;
+    }
+
+    /* Save to file */
+    hFile = CreateFileW(SavePath, GENERIC_WRITE, FILE_SHARE_READ, NULL, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, NULL);
+    if (hFile == INVALID_HANDLE_VALUE)
+    {
+        return FALSE;
+    }
+    bSuccess = WriteFile(hFile, pData, dwSize, &dwWritten, NULL);
+    CloseHandle(hFile);
+    if (!bSuccess)
+    {
+        return FALSE;
+    } else if (dwWritten != dwSize)
+    {
+        trace("Extract resource failed, written size (%lu) is not actual size (%lu).\n", dwWritten, dwSize);
+        SetLastError(ERROR_INCORRECT_SIZE);
+        return FALSE;
+    }
+    return TRUE;
+}
+
+static VOID NTAPI DllLoadCallback(
+    _In_ ULONG NotificationReason,
+    _In_ PCLDR_DLL_NOTIFICATION_DATA NotificationData,
+    _In_opt_ PVOID Context)
+{
+    LONG lRet;
+    HMODULE* phNotifiedDllBase = Context;
+
+    /*
+     * Verify the data,
+     * NotificationData->Loaded and NotificationData->Unloaded currently are the same.
+     */
+
+    /* Verify the FullDllName and BaseDllName */
+    ok_eq_ulong(NotificationData->Loaded.Flags, 0UL);
+    lRet = RtlCompareUnicodeString(NotificationData->Loaded.FullDllName,
+                                   (PCUNICODE_STRING)&g_usDllPath,
+                                   TRUE);
+    ok_eq_long(lRet, 0L);
+    lRet = RtlCompareUnicodeString(NotificationData->Loaded.BaseDllName,
+                                   (PCUNICODE_STRING)&g_usDllName,
+                                   TRUE);
+    ok_eq_long(lRet, 0L);
+
+    /*
+     * Verify SizeOfImage and read SizeOfImage from PE header,
+     * make sure the DLL is not unmapped, the memory is still accessible.
+     */
+    ok_eq_ulong(NotificationData->Loaded.SizeOfImage,
+                RtlImageNtHeader(NotificationData->Loaded.DllBase)->OptionalHeader.SizeOfImage);
+
+    /* Reason can be load or unload */
+    ok(NotificationReason == LDR_DLL_NOTIFICATION_REASON_LOADED ||
+       NotificationReason == LDR_DLL_NOTIFICATION_REASON_UNLOADED, "Incorrect NotificationReason\n");
+    if (NotificationReason == LDR_DLL_NOTIFICATION_REASON_LOADED)
+    {
+        *phNotifiedDllBase = NotificationData->Loaded.DllBase;
+        InterlockedIncrement(&g_lDllLoadCount);
+    } else if (NotificationReason == LDR_DLL_NOTIFICATION_REASON_UNLOADED)
```

Comment:
```suggestion
    }
    else if (NotificationReason == LDR_DLL_NOTIFICATION_REASON_UNLOADED)
```
usually I am a fan of saving LineOfCode, but this formatting is a bit too much even for my taste. ;-)

Reply:
LOC is not a metric anyone should look at for stuff like this.

Reply:
Well for once, it's explained there: https://reactos.org/wiki/Coding_Style#Braces